kratos/mpi/includes/mpi_communicator.h

```diff
@@ -645,9 +645,9 @@ class MPICommunicator : public Communicator
         return true;
     }
 
-    bool AssembleNonHistoricalData(Variable<vector<array_1d<double,3> > > const& ThisVariable) override
+    bool AssembleNonHistoricalData(Variable<DenseVector<array_1d<double,3> > > const& ThisVariable) override
```

Comment:
can you cherry-pick the changes in this file to a separate PR?

Reply:
This PR moves the MPICommunicator from the core to the new library, and unfortunately I made the porting after the move... I can replicate the changes in the core if we see that this takes too long to review/merge, but I'd avoid it otherwise, since it would only serve to create a merge-conflict.